lib/src/dom/html.dart

```diff
@@ -4166,18 +4174,24 @@ class PopStateEvent implements Event {
 
 extension PopStateEventExtension on PopStateEvent {
   external JSAny? get state;
+  external bool get hasUAVisualTransition;
 }
 
 @JS()
 @staticInterop
 @anonymous
 class PopStateEventInit implements EventInit {
-  external factory PopStateEventInit({JSAny? state});
+  external factory PopStateEventInit({
+    JSAny? state,
+    bool hasUAVisualTransition,
```

Comment:
@srujzs - should we be generating this as `bool? hasUAVisualTransition`? `required bool hasUAVisualTransition`? I assume the `bool` param is effectively optional.

Reply:
https://github.com/w3c/webref/blob/b8cdf28506d6699133866dbe950ff20873d84218/ed/idl/html.idl#L2045

`external` members are weird. You're allowed to write non-nullable members with no default to communicate that JS has its own default. This is probably better than `bool?` because it disallows users from passing a `null` value.